### PR TITLE
Add a new `GetFuturePParams` query

### LINKED ISSUE #1204: [FEAT] - New ledger state queries
**Internal**

**Describe the feature you'd like**
We need to add some queries.

New Conway ledger state queries have been requested and already implemented in ledger: https://github.com/IntersectMBO/cardano-ledger/pull/4514

* [`queryProposals`](https://github.com/IntersectMBO/cardano-ledger/blob/411054e40b4e08350049e4eaffdf0cc5f73e9d91/libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs#L328-L334)
* and `queryRatifyState`](https://github.com/IntersectMBO/cardano-ledger/blob/411054e40b4e08350049e4eaffdf0cc5f73e9d91/libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs#L346C1-L347)

Also there is another query that has been implemented that should be added, which works for all eras, not only Conway:

* [`queryFuturePParams`](https://github.com/IntersectMBO/cardano-ledger/blob/411054e40b4e08350049e4eaffdf0cc5f73e9d91/libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs#L317-L321)

**Describe alternatives you've considered**

I've considered implementing them myself, but decided to create this ticket instead :wink: 

**Additional context / screenshots**

### PR DESCRIPTION
**Pull request by dnadales**
Closes #1204 

Depends on https://github.com/IntersectMBO/ouroboros-network/pull/4960

### CODE REVIEW

File: ouroboros-consensus-cardano/changelog.d/20240917_143437_damian.nadales.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
++### Breaking
++
++- Add `GetFuturePParams` query.
```
Comment (amesgen) [inline ouroboros-consensus-cardano/changelog.d/20240917_143437_damian.nadales.md:3]:
```suggestion
### Breaking

- Add `GetFuturePParams` query.
```

File: ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs [hunk 2]
```diff
@@ -308,6 +308,10 @@ data instance BlockQuery (ShelleyBlock proto era) :: Type -> Type where
     :: CG.ConwayEraGov era
     => BlockQuery (ShelleyBlock proto era) (CG.RatifyState era)
 
+  GetFuturePParams
+    :: CG.ConwayEraGov era
+    => BlockQuery (ShelleyBlock proto era) (Maybe (LC.PParams era))
```
Comment (amesgen) [inline ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs:313]: `GetFuturePParams` isn't Conway-specific, the `ConwayEraGov` constraint is not necessary.

File: ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Node.hs [hunk 3]
```diff
@@ -474,6 +474,23 @@ pattern CardanoNodeToClientVersion13 =
       :* Nil
       )
 
+-- | The hard fork enabled, and the Shelley, Allegra, Mary, Alonzo and Babbage
+-- and Conway eras enabled, using 'ShelleyNodeToClientVersion9' for the
```
Comment (jasagredo) [inline ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Node.hs:478]: This comment is outdated?

File: ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs (hunk 2 above)
Comment (dnadales) [reply ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs:313]: Good point. I removed it.

**Review by amesgen — APPROVED** [review]
Approving the part of this diff that isn't also part of #1262

**Pull request merged**